core/gtk: use command line supplied by the shell in notifications

jcollie opened the pull request:
Some shells use OSC 133;C to tell us what command is being executed. Use that information to enhance our notifications that a command has finished.

mitchellh reviewed (changes requested):
Couple comments. Still working my way through it.

mitchellh reviewed (changes requested):
Can you pull out the parser updates from the rest?

jcollie committed:
lib: fix lib.String compilation

jcollie committed:
core/gtk: use command line supplied by the shell in notifications

Some shells use OSC 133;C to tell us what command is being executed.
Use that information to enhance our notifications that a command has
finished.

jcollie commented:
Rebased after #11046 was merged.
